conans/client/conan_api.py

```diff
@@ -411,36 +411,50 @@ def config_install(self, item):
         from conans.client.conf.config_installer import configuration_install
         return configuration_install(item, self._client_cache, self._user_io.out, self._runner)
 
+    def _info_get_profile(self, install_folder, profile_name, settings, options, env):
+        cwd = os.getcwd()
+
+        # If not install_folder we don't look in the curdir if any setting has specified
+        if not install_folder and (profile_name or settings or options or env):
```

Comment:
Not sure, why need to check ``install_folder``? Maybe it would be better to first discard error conditions, like ``validate_one_settings_source``, then, the logic would be simpler.

Reply:
Because the default of the install_folder is the curdir, so is probably that you find the info objects in the current directory, but you also has specified a `-s` for example. Should it raise? I think it doesn't because you didn't specify a ``--install-folder``

Reply:
This is what I mean. If the ``install_folder`` has as default the current dir, it is never False/None, then such check doesn't makes sense? What am I missing here? Probably if the function had only one ``profile_from_args`` call, the logic would be more understandable.

Reply:
Because before you default the `install_folder` to the `curdir` you have to check if there is some profile parameter to no default it. I'll try to refactor it and see what can be done.